### PR TITLE
DOC Amend `open_sample` docstring str->URI

### PR DESCRIPTION
**Pull request by lucyleeow**
# Fixes/Closes

# Description

Amend 'str' to 'URI' in `open_sample`, as we are moving towards npe2 (and generally more explicit)

# References

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (changes required to run napari, tests, & CI smoothly)
- [ ] Enhancement (non-breaking improvements of an existing feature)
- [ ] Documentation (update of docstrings and deprecation information)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by lucyleeow** [comment]
@jni

**Review by jni — APPROVED** [review]

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -965,8 +965,8 @@ def open_sample(
         sample : str
             name of the sample
         reader_plugin : str, optional
-            reader plugin to pass to viewer.open (only used if the sample data
-            is a string).  by default None.
+            reader plugin to use, passed to ``viewer.open`` (only used if the
+            sample data is a URI).  By default None.
```
Comment (GenevieveBuckley) [inline napari/components/viewer_model.py]: Probably a good idea to put the full definition of the URI acronym here too.

**Review by GenevieveBuckley — APPROVED** [review]

File: napari/components/viewer_model.py (hunk 1 above)
Comment (lucyleeow) [reply napari/components/viewer_model.py]: Thanks, done!

**Comment by GenevieveBuckley** [comment]
Thank you @lucyleeow

**Pull request merged**